### ISSUE TITLE
Is there a way to call a function after notifyListeners() finishes?

### ISSUE THREAD

**Issue by merfire** [Stacked-Org/stacked#105]
Is there a way to call a function after notifyListeners() finishes(after widget tree is fully rebuilt after the update)?

For example I have a list of objects each with TextEditingController and when I remove object from the list I want to dispose of the controller. Right now I remove the object, call notifyListeners(), then use Future.delayed(Duration(milliseconds: 100)) and then despose of the controller.

Maybe a simpler example of my problem would be if I want to add new TextField to the list and switch focus scope to it.
I would add TextField to the list, call notifyListeners() and then I can't just use FocusScope.of(context).nextFocus(), I need Future.delayed before calling FocusScope. 

Is there a better way of doing this and avoiding use of Future.delayed?

**Comment by RodolfoSilva** [comment]
Hi @merfire, maybe this can help you:

```dart
WidgetsBinding.instance.addPostFrameCallback((_) async {
  // You logic here....
});
```

**Comment by nahid18** [comment]
> Hi @merfire, maybe this can help you:
> 
> ```dart
> WidgetsBinding.instance.addPostFrameCallback((_) async {
>   // You logic here....
> });
> ```

Where to put your code in a stateless view? or maybe in the viewmodel?

**Comment by RodolfoSilva** [comment]
In the ViewModel @nahid18

**Comment by FilledStacks** [comment]
@merfire I would make all that automatic instead of manually doing it. Make a stateful widget that wraps a TextField and pass all the values you need + the controller. Use that when displaying your TextField and then call dispose on the controller in the dispose function that'll be called when it's removed from the widget tree.

**Comment by merfire** [comment]
Thanks to all for the answers, I tested both solutions, addPostFrameCallback solved my problem.

For anyone that has the same problem I just added addPostFrameCallback in ViewModel function like this:

ChecklistItemData is object that has controller(TextEditingController) 
items is List<ChecklistItemData>
deleteItem is a function in my ViewModel

``` 
void deleteItem(ChecklistItemData item) {
    items.remove(item);
    notifyListeners();
    WidgetsBinding.instance.addPostFrameCallback((_) async {
      item.controller.dispose();
    });
  }
```

The second solution with converting widget to stateful and disposing controller in dispose() function didn't work for me. Maybe I did something wrong, I am not sure, but let me explain what happened in this case.

I have a list of TextFields like todo app that can be deleted. Let's say I add 4 TextFields with text One, Two, Three, Four.
If I delete Four, dispose of controller Four will be called and everything is ok.
But if I delete Two, for some reason dispose will be called on Two, Three and Four.
After rebuild (notifyListeners) only TextField One has controller, Three and Four controllers are desposed (they are still in the items list and they do render, but without controller) and flutter throws an error: A TextEditingController was used after being disposed.
It seems like even if I am removing 1 item from my items list, all items after the deleted one are disposed too.
